python/ray/autoscaler/_private/monitor.py

```diff
@@ -181,7 +181,11 @@ def update_load_metrics(self):
             pending_placement_groups = list(
                 resources_batch_data.placement_group_load.placement_group_data)
 
-            ip = resource_message.node_manager_address
+            use_node_id_as_ip = self.autoscaler.config['provider'].get('init_cmd_run_by_scheduler_not_ray_NodeUpdater', False)
+            if use_node_id_as_ip:
+                ip = str(int(total_resources.get('NODE_ID_AS_RESOURCE', 0)))
+            else:
+                ip = resource_message.node_manager_address
```

Comment:
@wuisawesome @rkooo567  ignoring the actual implementation diff for now,

is it possible to switch the load metrics key from "ip" to "node id" (or make this configurable)?

Currently, @yuduber's use case prevents their cluster from having unique IP addresses for each "ray node".

Reply:
It's easy to make it Ray's node id, but I'm not sure how we'll be able to correlate that with the node provider's concept of a node.

A related question I'd have, is how does peloton distinguish between different nodes?

Reply:
Right now, peloton has both some environment variables set per "node", and I think some apps identify each node by correlating with a specific "<ip>:<port>" combo. 

@yuduber FYI

Reply:
in Peloton, we have the concept of task, one task is one container that will be provisioned to some machine.  To run Ray in peloton, we use one task to run one Ray node (head or worker node).  Each task has unique task_id and task_name and peloton can use either to distinguish different tasks.  My implementation used task_name, which is just a integer incremented from 0, where 0 represents head node, and 1 the nest worker node, etc.

The ip won't work here because one machine will have a unique ip, and it may have several containers provisioned on it.  So these containers will share the same ip address.

Reply:
This looks fine to me -- just name the field "use_node_id_as_ip" in the provider config and test for that.

Reply:
Sure.